app/src/main/java/net/gsantner/opoc/ui/SearchOrCustomTextDialog.java

```diff
@@ -195,13 +208,34 @@ protected FilterResults performFiltering(final CharSequence constraint) {
         }
     }
 
+    private static void setNeutralButton(final AlertDialog dialog, final WithPositionAdapter adapter) {
+        final Button button = dialog.getButton(AlertDialog.BUTTON_NEUTRAL);
+        if (button != null) {
+            if (adapter._dopt.multiSelectCallback != null && adapter._selected.size() > 0) {
+                button.setVisibility(Button.VISIBLE);
+                button.setText(R.string.clear);
+                button.setOnClickListener((v) -> {
+                    adapter._selected.clear();
+                    adapter.notifyDataSetChanged();
+                    setNeutralButton(dialog, adapter); // Will not result in recursion
+                });
+            } else if (adapter._dopt.neutralButtonCallback != null && adapter._dopt.neutralButtonText != 0) {
+                button.setVisibility(Button.VISIBLE);
+                button.setText(adapter._dopt.neutralButtonText);
+                button.setOnClickListener((v) -> adapter._dopt.neutralButtonCallback.callback());
+            } else {
+                button.setVisibility(Button.INVISIBLE);
+            }
+        }
+    }
```

Comment:
Biggest change is that if multiple items are selected, the neutral button becomes a 'clear' button. It will revert to the specified button once clear has been pressed (or all items manually deselected).